### PR TITLE
feat: [Fullscreen Menu] Integrate current features into the new fullscreen menu

### LINKED ISSUE #1431: [Explore V2] Highlights sub-section implementation
## What?
- [x] Create the **Highlights** sub-section in the Explore V2 feature.

## Related links
- **PRD "Explore V2"**: https://www.notion.so/decentraland/PRD-Explore-V2-ab98e922c196488fb217596c888ce3d6

### LINKED ISSUE #1520: Integrate Explore into Fullscreen Menu
- [x] Integrate Explore into Fullscreen Menu

### LINKED ISSUE #1521: Integrate Backpack into Fullscreen Menu
- [x] Add backpack integration to fullscreen menu.
- [x] Remove backpack music (not for avatar creation workflow)
- [x] On pressing "I", the Fullscreen menu opens directly with the backpack section selected
- [x] Done button saves the changes and close the Fullscreen Menu

### LINKED ISSUE #1522: Polish Fullscreen Menu header
- [x] On keypress TAB open Fullscreen Menu on the previously selected section
- [ ] On keypress ESC on the last level (no UI, no cursor lock) opens Fullscreen Menu on the previously selected section
- [x] On keypress ESC or TAB with the Fullscreen menu opened, closes the Menu
- [x] Add Realm name to Decentraland Logo and remove number of players
- [x] Bring profile widget on clicking the profile picture
- [ ] Add a new Button to launch Fullscreen Menu

### LINKED ISSUE #1523: Integrate Map into Fullscreen Menu
- [x] On key press M opens the Fullscreen Menu on the Map section
- [x] TAB does not open the Map section anymore
- [x] Clicking on the Minimap opens the Fullscreen Menu on the Map section
- [x] Remove close button from the current Map UI
- [x] On Jump In while a scene is selected, should teleport and close the Fullscreen Menu

### LINKED ISSUE #1524: Integrate Builder into Fullscreen Menu
- [x] Integrate Builder panel as it is right now in a new section Builder inside Fullscreen Menu
- [x] Interacting with any of the actions of this panel (Edit, Jump In etc) closes the Fullscreen Menu
- [x] Remove button of Builder from the Taskbar
- [x] Consider Builder section show/hide based on Feature Flag

### LINKED ISSUE #1525: Integrate Settings into Fullscreen Menu
- [x] Integrate settings section as it is in a new Settings section inside the Fullscreen Menu
- [x] On pressing P open the fullscreen menu on the Settings section

### LINKED ISSUE #1526: Integrate Quests into Fullscreen Menu
- [x] Integrate Quests section as it is
- [x] Remove button of Quest from the Taskbar
- [x] Consider Quests section show/hide based on Feature Flag

### LINKED ISSUE #1527: Integrate More Actions into Fullscreen Menu
- [x] Switch Day/Night will be replaced with procedural skybox, in the meantime it could be added to the settings
- [x] Show/hide Taskbar no longer makes sense as the taskbar will be gone
- [x] Hide UI should be integrated in the settings
- [x] Hide Names should be integrated in the settings
- [x] Tutorial could be integrated in the Settings section
- [x] Report a bug could be integrated in the Settings section
- [x] Help & Support could be integrated in the Settings section
- [x] Delete more actions button

### LINKED ISSUE #1559: Add Fullscreen Menu explanation to the Tutorial
- [x] Move the Users Around step after the Voice Chat one.
- [x] Change tooltip for opening Start Menu.
- [x] On Start Menu open:
   - [x] Generic message explaining.
   - [x] If user clicks on each section, specific tooltips will be open.
- [x] ProfileHUD:
   - [x] Remove backpack button.
   - [x] Update tooltip message.
- [x] Use TAB for navigating between sections.

### LINKED ISSUE #1594: [Fullscreen Menu] Visual refinements
Apply all the needed visual refinements.

**NOTE**: Please use the branch `feat/ExploreV2-Highlights-sub-section` to implement the changes.

- [x] Update `AvatarEditorHUD` to be adapted to the fullscreen menu correctly (Santi).
- [x] Update the tutorial flow according with the lastest conversations (Santi).
- [x] Update the `ControlsHUD` with the new shorcuts (Romi).
- [x] Background image tiling (Romi).
- [x] Decentraland icon border (Romi).
- [x] Update the Explore section layout with the new fullscreen config (Romi).
- [x] Update the header sections icons + color bar (Romi).
- [x] Fullscreen menu enter / exit animation (Stefi / Santi).
- [x] Fullscreen menu button idle / hover / unhover / click animation (Stefi / Santi).
- [x] Taskbar background (Romi).
- [x] Tutorials keyboard animated images (press T, ESC) (Romi).
- [x] Add new configuration to the sections selector UI component in order to set different icons on selected/unselected state (Santi)

### LINKED ISSUE #1610: [Fullscreen Menu] Refinement tasks
- [x] Add sound for each section opening.
- [x] When the tutorial is in progress and open the Start Menu, set the Explore section as default.
- [x] Unify the tutorial steps "social" and "users around".
- [x] Remove the minimap tutorial step.
- [x] Set a default image for cards that were failed during its image loading.
- [x] Move Start menu button from taskbar to the minimap.
- [x] Remove all right side of the taskbar (it will be used only for Portable Experiences).
- [x] Review and update all tutorial messages.
- [x] The "Show Avatar Names" setting is being always loaded as `false` (it is only happening in build, not in editor).
- [x] Review the current analytics.
- [x] Add tests for all the new implementation.

### PR DESCRIPTION
**Pull request by sandrade-dcl**
### Integration of the current features to the new fullscreen menu
- [x] Fix #1522 
- [x] Fix #1520 
- [x] Fix #1521 
- [x] Fix #1523
- [x] Fix #1524
- [x] Fix #1525
- [x] Fix #1526
- [x] Fix #1527
- [x] Fix #1559 
- [x] Fix #1610 
- [x] Fix #1594 

### Implementation of the new Highlights sub-section
- [x] Fix #1431

## Screenshots
image
image
image
image
image
image

## How to test the changes?
1. Go to: https://play.decentraland.zone/?renderer-branch=feat/ExploreV2-Highlights-sub-section (ORG) or https://play.decentraland.zone/?renderer-branch=feat/ExploreV2-Highlights-sub-section&NETWORK=ropsten (ROPSTEN)
2. Open the new Fullscreen Menu from the taskbar.
3. Test all features included are working correctly.

## Dependencies
Before merge this PR, wait for the merge of https://github.com/decentraland/unity-renderer/pull/1629.

## Useful Links
**PRD**: https://www.notion.so/decentraland/PRD-Explore-V2-ab98e922c196488fb217596c888ce3d6
**Figma**: https://www.figma.com/file/5YhRkm1NXme8MhgDxGT2zY/ExploreV2?node-id=247%3A0

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Comment by Suduck** [comment]
Good Job! Just smaill things that I noted, I will continue with the code review

- (Very small thing) If you go to the map , look for some parcel, and them you try to move to another tab, the map will try to fetch the info of the last parcel where the mouse was before leaving to another tab 
- Settings view is a little weird now since we have 2 columns but a lot of space
-Info of the events Pop up in the explore is behind the top bar  (Screenshot added)
-The icon is above the mini-map and it is a bit weird is this intended?
-The background of the avatar backpack is different from the explorer or builder, is that intended?

image

**Comment by Suduck** [comment]
Also, I am unable to see the shortcuts or keybinding no more

File: unity-renderer/Assets/DCLPlugins/BuilderInWorld/HUD/ProjectsPanelHUD/Scripts/BuilderMainPanelController.cs [hunk 1]
```diff
@@ -55,12 +54,17 @@ public class BuilderMainPanelController : IHUD, IBuilderMainPanelController
 
     internal IContext context;
 
+    BaseVariable<Transform> configureBuilderInFullscreenMenu => DataStore.i.exploreV2.configureBuilderInFullscreenMenu;
```
Comment (Suduck) [inline unity-renderer/Assets/DCLPlugins/BuilderInWorld/HUD/ProjectsPanelHUD/Scripts/BuilderMainPanelController.cs:57]: Do we need this? 

Is any situation where we won't show the panel in fullscreen?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/EventsSubSection/EventsSubSectionMenu/EventsSubSectionComponentController.cs [hunk 2]
```diff
@@ -361,8 +270,6 @@ internal void SubscribeToEvent(string eventId)
         //    {
         //        Debug.LogError($"Error posting 'attend' message to the API: {error}");
         //    });
```
Comment (Suduck) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/EventsSubSection/EventsSubSectionMenu/EventsSubSectionComponentController.cs:272]: remove commented code

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/EventsSubSection/EventsSubSectionMenu/EventsSubSectionComponentController.cs [hunk 3]
```diff
@@ -382,7 +289,5 @@ internal void UnsubscribeToEvent(string eventId)
...
```
Comment (Suduck) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/EventsSubSection/EventsSubSectionMenu/EventsSubSectionComponentController.cs:291]: remove commented code

**Review by Suduck — APPROVED** [review]
Good job! Small requests

**Comment by sandrade-dcl** [comment]
Thanks for the feedback, Adri! I will answer each question below:

> * (Very small thing) If you go to the map , look for some parcel, and them you try to move to another tab, the map will try to fetch the info of the last parcel where the mouse was before leaving to another tab

I think I have not understand the issue correctly, I will ask you tomorrow in order to reproduce it!

> * Settings view is a little weird now since we have 2 columns but a lot of space

Yeah, for now it is a very first version of the Start Menu where we decided to move the different features directly inside the menu as they were. Most of these feature are planned to be re-designed in the future.

Other detail: take into account that this PR is still pending to be touched by Romi this week in order to adjust several visual details along all the Start Menu.

>   -Info of the events Pop up in the explore is behind the top bar  (Screenshot added)

It is the shape of the modal! The upper part coincides exactly with the menu header, but there is not any part of it that is behind. Anyway these kind of details will also be reviewed by Romi.

>   -The icon is above the mini-map and it is a bit weird is this intended?

It is pending to be modified by Romi. The idea will be to separate the button from the minimap with a noticeable border.

>   -The background of the avatar backpack is different from the explorer or builder, is that intended?

Pending to be modified by Romi.

> * I see DG on the top left but I don't see any reference to realm, it seems like we are DG :D

At the moment it is the current design of the "realm visualizator". Anyway lets Romi to decide if she wants to update it after your feedback, thanks!

> * Hide UI setting is working strange, if you click on it the UI should disappear ( include the fullscreen menu)

It is something intended, we don't want to hide the Start Menu with this setting but only the UI that is visible during the "world view".

> * I am unable to see the shortcuts or keybinding no more

Now , in the new design, the `ControlsHUD` is only accessible through the `C` shortcut (it is something that is explained during the tutorial). Anyway, maybe it would be a good ide to add a "CONTROLS" button in the Settings section to be accessible by the users. Lets Romi decide!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/EventsSubSection/EventsSubSectionMenu/EventsSubSectionComponentController.cs (hunk 2 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/EventsSubSection/EventsSubSectionMenu/EventsSubSectionComponentController.cs:282]: For the moment I prefer to leave it like this (it has a `TODO` mark) in order to have the code reference when we implement the "Subscribe" API call in the future.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/EventsSubSection/EventsSubSectionMenu/EventsSubSectionComponentController.cs (hunk 3 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/EventsSubSection/EventsSubSectionMenu/EventsSubSectionComponentController.cs:301]: For the moment I prefer to leave it like this (it has a `TODO` mark) in order to have the code reference when we implement the "Unsubscribe" API call in the future.

File: unity-renderer/Assets/DCLPlugins/BuilderInWorld/HUD/ProjectsPanelHUD/Scripts/BuilderMainPanelController.cs (hunk 1 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/DCLPlugins/BuilderInWorld/HUD/ProjectsPanelHUD/Scripts/BuilderMainPanelController.cs:57]: The Builder panel HUD must wait to receive the specific `Transform` (from the `ExploreV2` feature) where it has to be parented. This transform is not fully fullscreen but is part of a specific visual area below the menu header. In this waym the builder panel will be re-adapted to the correct place in the correct moment.

**Pull request merged**